drivers/CoreInterruptIn/include/CoreInterruptIn.h

```diff
@@ -0,0 +1,29 @@
+// Leka - LekaOS
+// Copyright 2023 APF France handicap
+// SPDX-License-Identifier: Apache-2.0
+
+#pragma once
+
+#include "CoreEventQueue.h"
+#include "interface/drivers/InterruptIn.h"
+
+namespace leka {
+
+class CoreInterruptIn : public interface::InterruptIn
+{
+  public:
+	explicit CoreInterruptIn(PinName pin) : _interrupt_in(pin) { _event_queue.dispatch_forever(); };
+
+	auto read() -> int final;
+
+	void rise(std::function<void()> const &callback) final;
+	void fall(std::function<void()> const &callback) final;
+
+  private:
+	CoreEventQueue _event_queue;
+	mbed::InterruptIn _interrupt_in;
```

Comment:
as there is only one pin:

```suggestion
	explicit CoreInterruptIn(PinName pin) : _pin(pin) { _event_queue.dispatch_forever(); };

	auto read() -> int final;

	void rise(std::function<void()> const &callback) final;
	void fall(std::function<void()> const &callback) final;

  private:
	CoreEventQueue _event_queue;
	mbed::InterruptIn _pin;
```